src/store/middlewares/roomMiddleware.tsx

```diff
@@ -53,8 +54,14 @@ const createRoomMiddleware = ({
 									);
 									dispatch(webrtcActions.setIceServers(turnServers));
 									dispatch(webrtcActions.setRTCStatsOptions(rtcStatsOptions));
+									// dispatch(webrtcActions.setClientMonitorConfig(clientMonitorConfig));
 									dispatch(roomActions.setState('joined'));
 									dispatch(joinRoom());
+
+									const roomId = getState().room.name;
```

Comment:
These are not state actions, please move them out of the batch call, either before or after doesn't matter.